File: frontend/src/types/dto/entity/GameStateDto.ts
import type { PlayerColour } from "@/types/enum/PlayerColour"
import type { UserDto } from "./UserDto"

export type GameStateDto = {
    id: number,
    gameId: number,
    user: UserDto,
    score: number,
    playerColour: PlayerColour,
    currentBoardState: string,
    // 16 bit array representing bitmask of each cell's notes
    notes: Uint16Array
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/game/GameStateDto.java
package com.github.ryand6.sudokuweb.dto.entity.game;

import com.github.ryand6.sudokuweb.dto.entity.user.UserDto;
import com.github.ryand6.sudokuweb.enums.PlayerColour;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GameStateDto {

    private Long id;

    private Long gameId;

    private UserDto user;

    private Integer score;

    private PlayerColour playerColour;

    private String currentBoardState;

    private byte[] notes;

}
